client/state/logstash/actions.ts

```diff
@@ -26,9 +25,6 @@ interface LogToLogstashParams {
  * @param params wpcom log2logstash params.
  * @returns      Action object
  */
-export function logToLogstash( params: LogToLogstashParams ) {
-	return {
-		type: LOGSTASH,
-		params: { params: JSON.stringify( params ) },
-	};
-}
+export const logToLogstash = ( params: LogToLogstashParams ) => {
```

Comment:
Let's move this one out of `/state` - it's no longer related to Redux. `/lib` looks like a good place for the thunk.

Reply:
I'm considering removing the Redux action completely. And just import a `logToLogstash` function from a `/lib` module and call it. Without any `dispatch` indirection or other bells and whistles. It would probably require some extra mocking in tests but otherwise I guess it's fine? Let me know what you think.

Reply:
Yeah, I think we're talking about the same thing, that's what I was essentially suggesting.

Reply:
OK, did that in 395e73c. Makes the PR a lot bigger. The toughest module was `composite-checkout/lib/analytics`. There, I refactored some actions to thunks. The idea is that when a function takes `dispatch` as parameter:
```js
recordTransactionBeginAnalytics( { paymentMethodId, reduxDispatch } );
```
it should really be a thunk that is dispatched:
```js
reduxDispatch( recordTransactionBeginAnalytics( { paymentMethodId } ) );
```
and defined as:
```js
const recordTransactionBeginAnalytics = ( { paymentMethodId } ) => ( dispatch ) => ...
```
Cc @sirbrillig for review of this part.

Reply:
I haven't had a chance to test it yet, but the changes look sensible to me. As long as these events are still being recorded, it should be fine.